source/agora/node/Ledger.d

```diff
@@ -585,8 +585,12 @@ public class Ledger
         @safe
     {
         data = ConsensusData.init;
-        data.timestamp = max(clock.networkTime(), this.last_block.header.timestamp + 1);
-        log.trace("Going to nominate current timestamp [{}] or newer", clock.networkTime());
+        const ulong genesisTimestamp = this.params.GenesisTimestamp;
+        const ulong now = clock.networkTime;
+        const ulong genesis_offset = now - genesisTimestamp;
```

Comment:
What about underflow ?

Reply:
Updated to return early and log error if time is before Genesis timestamp.